test/pg-cdc/alter-table-after-source.td

```diff
@@ -0,0 +1,407 @@
+INSERT INTO remove_column VALUES (3);
+
+! SELECT * from remove_column;
+contains:altered
+
+> DROP VIEW remove_column;
+> DROP SOURCE mz_source;
+
+$ postgres-execute connection=postgres://postgres:postgres@postgres
+ALTER USER postgres WITH replication;
+DROP SCHEMA IF EXISTS public CASCADE;
+DROP PUBLICATION IF EXISTS mz_source;
+
+CREATE SCHEMA public;
+
+CREATE TABLE alter_column (f1 INTEGER, f2 VARCHAR(2));
+ALTER TABLE alter_column REPLICA IDENTITY FULL;
+INSERT INTO alter_column VALUES (2, 'ab');
+
+CREATE PUBLICATION mz_source FOR ALL TABLES;
+
+> CREATE MATERIALIZED SOURCE mz_source
+  FROM POSTGRES CONNECTION 'host=postgres port=5432 user=postgres password=postgres sslmode=require dbname=postgres'
+  PUBLICATION 'mz_source';
+
+> CREATE VIEWS FROM SOURCE mz_source;
+
+> SELECT * from alter_column;
+2 ab
+
+$ postgres-execute connection=postgres://postgres:postgres@postgres
+ALTER TABLE alter_column ALTER COLUMN f2 TYPE CHAR(2);
+INSERT INTO alter_column VALUES (3, 'bc');
+
+! SELECT * from alter_column;
+contains:altered
+> DROP VIEW alter_column;
+> DROP SOURCE mz_source;
+
+$ postgres-execute connection=postgres://postgres:postgres@postgres
+ALTER USER postgres WITH replication;
+DROP SCHEMA IF EXISTS public CASCADE;
+DROP PUBLICATION IF EXISTS mz_source;
+
+CREATE SCHEMA public;
+
+CREATE TABLE alter_nullability (f1 INTEGER NOT NULL);
+ALTER TABLE alter_nullability REPLICA IDENTITY FULL;
+INSERT INTO alter_nullability VALUES (1);
+
+CREATE PUBLICATION mz_source FOR ALL TABLES;
+
+> CREATE MATERIALIZED SOURCE mz_source
+  FROM POSTGRES CONNECTION 'host=postgres port=5432 user=postgres password=postgres sslmode=require dbname=postgres'
+  PUBLICATION 'mz_source';
+
+> CREATE VIEWS FROM SOURCE mz_source;
+
+> SELECT * from alter_nullability
+1
+
+$ postgres-execute connection=postgres://postgres:postgres@postgres
+ALTER TABLE alter_nullability ALTER COLUMN f1 DROP NOT NULL;
+INSERT INTO alter_nullability VALUES (NULL);
+
+# ! SELECT * from alter_nullability;
+# contains:altered
+
+> DROP VIEW alter_nullability;
+> DROP SOURCE mz_source;
+
+$ postgres-execute connection=postgres://postgres:postgres@postgres
+ALTER USER postgres WITH replication;
+DROP SCHEMA IF EXISTS public CASCADE;
+DROP PUBLICATION IF EXISTS mz_source;
+
+CREATE SCHEMA public;
+
+CREATE TABLE alter_drop_pk (f1 INTEGER PRIMARY KEY);
+ALTER TABLE alter_drop_pk REPLICA IDENTITY FULL;
+INSERT INTO alter_drop_pk VALUES (1);
+
+CREATE PUBLICATION mz_source FOR ALL TABLES;
+
+> CREATE MATERIALIZED SOURCE mz_source
+  FROM POSTGRES CONNECTION 'host=postgres port=5432 user=postgres password=postgres sslmode=require dbname=postgres'
+  PUBLICATION 'mz_source';
+
+> CREATE VIEWS FROM SOURCE mz_source;
+
+> SELECT * from alter_drop_pk
+1
+
+$ postgres-execute connection=postgres://postgres:postgres@postgres
+ALTER TABLE alter_drop_pk DROP CONSTRAINT alter_drop_pk_pkey;
+INSERT INTO alter_drop_pk VALUES (1);
+
+# ! SELECT * from alter_drop_pk;
```

Comment:
If the DROP CONSTRAINT is not going to result in an error, then this part of the test should read:
```
> SELECT DISTINCT f1 FROM alter_drop_pk;
1
1
```